### ISSUE TITLE
Add support to enable descendants when creating/editing a BIE.

### ISSUE THREAD

**Issue by jim-wilson-kt** [OAGi/Score#1339]
<img width="853" alt="image" src="https://user-images.githubusercontent.com/2702530/194683328-b1c77ca5-80f4-40aa-bb6b-aff8d173938c.png">

**Comment by jim-wilson-kt** [comment]
This will be a key feature for the agriculture industry for their field operations data model.

**Comment by kbserm** [comment]
I would want to understand the nature of the requirement more. But another idea comes to mind is to have an inverse mode in BIE editing. That is if most fields will be used, instead of the users enabling nodes, they will disable nodes instead.

**Comment by jim-wilson-kt** [comment]
Yes. If I had to choose between the two, I would choose the inverse mode.

Regarding the requirement, the ADAPT Standard is very domain-specific. (The ADAPT Standard is what AgGateway is developing in Score and plans to release in some form on 2022-11-15.) AgGateway intends on publishing the entire model as JSON Schema. Whether that wll be a single file or set of files is TBD.

**Comment by kbserm** [comment]
Oh.... @hakjuoh will have a standalone generation though.

**Comment by hakjuoh** [comment]
@kbserm
I'm trying to test this inverse mode in the BIE, and I found some circular reference cases like

Trademark -> Device Resource -> Device Item -> Trademark
Metadata -> Software Resource -> Software Item -> Trademark -> Mark Attachment -> Metadata

The current logic can't express the BIE with this circular reference since it's been using the `Russian Doll` pattern. Even if it eliminates circular references from the expression, the size of the BIE is too big in some cases (e.g., the `Process Purchase Order` BIE expression in XML is 888.7Mb.)

We may need some mixed patterns like local declarations for BIEs and global declarations for the rest. Any idea?

**Comment by jim-wilson-kt** [comment]
I'm OK with stopping at the first level of a circular reference.

**Comment by kbserm** [comment]
That's why if exporting the whole thing it has got to be one of the standalone patterns (venetian blinds or garden of enden or salami slice) not Russian doll.

**Comment by dubnemo** [comment]
The only use case I have seen was for the ADAPT instance, where AgGateway
started with an *empty *CCL except for the CDT and select BDTs.

Scott

**Comment by jim-wilson-kt** [comment]
Agree with @dubnemo. I should point out that AgGateway will not be expressing BIEs as XSD. They plan to express as JSON and I can imagine OAS. While exporting "the whole thing" might be useful, there will still be a case for "select all descendants" or "inverse mode". This is relevant because the ADAPT Standard domain is very tight and therefore much more (perhaps all) of a selected structure would be in scope for the specified context.

**Comment by hakjuoh** [comment]
@kbserm @jim-wilson-kt 
Here's the 'Inverse Mode' implementation. In this example, the new BIE 'Financial Account Reference' has no 'Used' BIE, which means it has no DB records in Score.
<img width="615" alt="Screen Shot 2023-01-05 at 10 06 46 AM" src="https://user-images.githubusercontent.com/16102116/210812441-6dee8e90-6300-45b3-a7d4-fef64bd2a258.png">

If the user turns the 'Inverse Mode' switch on, the tree looks like all BIEs are checked. But, there are still no DB records for checked BIEs.
<img width="600" alt="Screen Shot 2023-01-05 at 10 07 00 AM" src="https://user-images.githubusercontent.com/16102116/210812966-1319efb0-249d-46e4-ba79-4f2847344997.png">

I should note that the 'Uplift BIE' function does not work well for the 'Inverse Mode' BIEs. Because it requires 'Used' flagged-DB records.

**Comment by dubnemo** [comment]
@jim-wilson-kt @hakjuoh 

I am not convinced the Inverse Mode is the right answer, since when you want to save the enabled properties, the request will be too large.  

I would like to share what Kelly Nelson shared with us:

**BIE Creation.**  As we’ve previously identified, publishing a very large BIE within Score has problems due to intentional constraints within the UI; the BIE facility is intended for smaller, narrower use items.  One needs to expand all children and click “enable children” on each to include the entities in the BIE.   Since this was a test run, I didn’t fret too much if I accidentally omitted something but tried to include everything.   Once the initial version is done, conceivably one can go back carefully and fine tune things without the initial time investment.

- The first pass at “enabling children” took 20 minutes.  When I got done, I clicked Update and promptly got a **413 Request Too Large Error.**
- I uttered a few choice words and tried it again, this time clicking Update frequently as I went.   This time it worked.
